hardware_reservations.go

```diff
@@ -0,0 +1,100 @@
+package packngo
+
+import "fmt"
+
+const hardwareReservationBasePath = "/hardware-reservations"
+
+// HardwareReservationService interface defines available hardware reservation functions
+type HardwareReservationService interface {
+	Get(hardwareReservationID string, listOpt *ListOptions) (*HardwareReservation, *Response, error)
+	List(projectID string, listOpt *ListOptions) ([]HardwareReservation, *Response, error)
+	Move(string, string) (*HardwareReservation, *Response, error)
+}
+
+// HardwareReservationServiceOp implements HardwareReservationService
+type HardwareReservationServiceOp struct {
+	client *Client
+}
+
+// HardwareReservation struct
+type HardwareReservation struct {
+	ID        string    `json:"id,omitempty"`
+	ShortID   string    `json:"short_id,omitempty"`
+	Facility  Facility  `json:"facility,omitempty"`
+	Plan      Plan      `json:"plan,omitempty"`
+	Href      string    `json:"href,omitempty"`
+	Project   Project   `json:"project,omitempty"`
+	Device    Device    `json:"device,omitempty"`
+	CreatedAt Timestamp `json:"created_at,omitempty"`
+}
+
+type hardwareReservationRoot struct {
+	HardwareReservations []HardwareReservation `json:"hardware_reservations"`
+	Meta                 meta                  `json:"meta"`
+}
+
+// List returns all hardware reservations for a given project
+func (s *HardwareReservationServiceOp) List(projectID string, listOpt *ListOptions) (reservations []HardwareReservation, resp *Response, err error) {
+	root := new(hardwareReservationRoot)
+	var params string
+	if listOpt != nil {
+		params = listOpt.createURL()
+	}
+
+	path := fmt.Sprintf("%s/%s%s?%s", projectBasePath, projectID, hardwareReservationBasePath, params)
+
+	for {
+		subset := new(hardwareReservationRoot)
+
+		resp, err = s.client.DoRequest("GET", path, nil, root)
+		if err != nil {
+			return nil, resp, err
+		}
+
+		reservations = append(reservations, root.HardwareReservations...)
+
+		if subset.Meta.Next != nil && (listOpt == nil || listOpt.Page == 0) {
+			path = subset.Meta.Next.Href
+			if params != "" {
+				path = fmt.Sprintf("%s&%s", path, params)
+			}
+			continue
+		}
+
+		return
+	}
+}
+
+// Get returns a single hardware reservation
+func (s *HardwareReservationServiceOp) Get(hardwareReservationdID string, listOpt *ListOptions) (*HardwareReservation, *Response, error) {
```

Comment:
The Get doesn't work on the API side:
https://github.com/packethost/packngo/issues/102